common/repo/tests/MemoryRepo.java

```diff
@@ -78,6 +78,10 @@ public void switchBranch(String branch) {
     currentBranch = branch;
   }
 
+  public void tagHead(String name) {
+    throw new UnsupportedOperationException("Not implemented");
```

Comment:
@oferb how about making `HashMap<Commit, String> tags` to support it?

Reply:
I'd rather not invest in MemoryRepo, until we actually start using it.